### DISCUSSION TITLE
Highlighting Syntax syntax?

### DISCUSSION THREAD

**Discussion by generrosity** [github-linguist/linguist#5679] [Q&A]
Hello!

I've been trying to find a way to display the syntax that a programmer might describe their shell program with. Appologies - its a bit of a meta topic, I might not be using quite the right working, and I couldn't find anything touching on it.

Here are samples of how people describe require, optional, alternative, string, and repeditive parameters. Is this something already implimented somewhere, or possibly missed?

Sample from ss64;
![image](https://user-images.githubusercontent.com/12838061/145108265-124dd409-cb1b-4e6f-bdd4-92e7c8b50ba2.png)

Sample from Powershell;
![image](https://user-images.githubusercontent.com/12838061/145108437-d598cea9-296f-4e29-a732-c6dea679d651.png)

Sample from Linux MAN;
![image](https://user-images.githubusercontent.com/12838061/145108181-d4d222bf-858e-4722-8b60-b1fc5226a89f.png)

Sample from C# reference on IF syntax;
![image](https://user-images.githubusercontent.com/12838061/145111230-085115ad-0513-4c18-8817-ef5ecf0b04a9.png)

Sample from Github;
![image](https://user-images.githubusercontent.com/12838061/145108223-42f431d5-4c82-475c-b8df-cc625d0f757b.png)
Sample from Github;
![image](https://user-images.githubusercontent.com/12838061/145108557-b63fbf1b-cbe5-4519-9541-b0ad6dad4857.png)

Obviously its not possible to trigger Liguist to work on inline text (or fool it to do so), and it doesn't appear possible to customize text within a code fence. Using 'vim' error highlighting isn't the best workaround.
![image](https://user-images.githubusercontent.com/12838061/145109904-c8f8c235-1c2d-4ded-a531-58ec3844f849.png)

Thanks for your patience and tips!

**Comment by lildude** [comment]
> I've been trying to find a way to display the syntax that a programmer might describe their shell program with.

Where are you trying to do this? 

From this… 

> Obviously its not possible to trigger Liguist to work on inline text (or fool it to do so), and it doesn't appear possible to customize text within a code fence.

… can I assume here on GitHub? If so, you should be able to use [fenced code blocks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting); the trick is identifying the correct language 😉 

And then you're beholden to the upstream grammar for doing the right thing which may not be many as usage information is normally in comments (like in shell scripts) and almost certainly not written in the same language so may not be highlighted any differently from the other comments or code.

**Reply by generrosity** [reply]
Thanks for the comment! Yes Github fenced code blocks, and yes wondering if existing language supports this. 

I've experimented with a couple dozen languages, but they are dealing with code and comments inside files, rather than documentation outside of code. Often commands and APIs don't have user visible comments to guide them.

Even REST API comes to mind now too `VERB https://{instance}/{collection}/{team-project}/_apis/{area}/{resource}?api-version={version}` and helping to guide the eye
![image](https://user-images.githubusercontent.com/12838061/145316965-ac8d02e2-3799-4570-9234-a7d03d34eb05.png)

Would a meta-grammer like this be useful?

**Reply by lildude** [reply]
> Would a meta-grammer like this be useful?

It would, and we already have one. @Alhadis maintains [languages-etc](https://github.com/Alhadis/language-etc) which is used for quite a few languages already. This could be added but the challenge comes in what languages are these written and so people write files with just this content?

Linguist needs a language name in order to pick the right grammar for the syntax highlighting and as these aren't a specific language with files dedicated to the language, their identification by Linguist is going to be hard.

As this is for code blocks and doesn't involve an identifiable language for the syntax, I think this functionality may be better suited in the [markup](https://github.com/github/markup) repo which deals with the rendering of things like the README.md files.

What do you think @Alhadis?

**Reply by Alhadis** [reply]
> What do you think @Alhadis?

If I understand the OP correctly, they're asking for a grammar that can highlight any meta-syntactic notation used by authors to describe a plain-text format. If so, such a thing can't be done: there're simply too many conflicting variations in use by authors.

The only thing I _can_ suggest is whitelisting HTML classes that are used for syntax highlighting on GitHub. See github/markup#905.
